packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -188,90 +209,89 @@ Future<XcodeBuildResult> buildXcodeProject({
   );
   await processPodsIfNeeded(project.ios, getIosBuildDirectory(), buildInfo.mode);
 
+  return ParsedProjectInfo(
+    autoSigningConfigs: autoSigningConfigs,
+    projectInfo: projectInfo,
+    scheme: scheme,
+    configuration:configuration,
+    hasWatchCompanion: hasWatchCompanion,
+    workspace: workspace,
+  );
+}
+
+Future<XcodeBuildResult> buildXcodeProject({
+  @required BuildableIOSApp app,
+  @required BuildInfo buildInfo,
+  String targetOverride,
+  bool buildForDevice,
+  DarwinArch activeArch,
+  bool codesign = true,
+  String deviceID,
+}) async {
+  final ParsedProjectInfo parsedProjectInfo = await _valdateXcodeBuild(
+    app: app,
+    buildInfo: buildInfo,
+    codesign: codesign,
+    buildForDevice: buildForDevice,
+    targetOverride: targetOverride,
+    deviceID: deviceID,
+  );
+  if (parsedProjectInfo == null) {
+    return XcodeBuildResult(success: false);
+  }
   final List<String> buildCommands = <String>[
```

Comment:
I tried to collapse most of the logic for creating the xcodebuild command using if/for elements - it should be easier to read for the most part...